packages/playwright-core/src/grid/gridServer.ts

```diff
@@ -239,17 +263,23 @@ export class GridServer {
     this._wsServer.on('connection', async (ws, request) => {
       if (request.url?.startsWith(this._securePath('/claimWorker'))) {
         const params = new URL('http://localhost/' + request.url).searchParams;
-        if (params.get('pwVersion') !== this._pwVersion && !process.env.PWTEST_UNSAFE_GRID_VERSION) {
+        const version = params.get('pwVersion') || request.headers['x-playwright-version'];
```

Comment:
Where does `x-playwright-version` come from? I thought that client sends `UserAgent` instead.

Reply:
removed as we don't use it anyway, version story needs more thoughts